sdks/java/extensions/jackson/src/main/java/org/apache/beam/sdk/extensions/jackson/AsJsons.java

```diff
@@ -64,13 +77,75 @@ private AsJsons(Class<? extends InputT> outputClass) {
               @Override
               public String apply(InputT input) {
                 try {
-                  ObjectMapper mapper = Optional.fromNullable(customMapper).or(DEFAULT_MAPPER);
-                  return mapper.writeValueAsString(input);
+                  return writeValue(input);
                 } catch (IOException e) {
-                  throw new RuntimeException(
+                  throw new UncheckedIOException(
```

Comment:
We could pass a `Contextful` here, in which case we'd be able to let the `IOException` bubble up rather than catching and rethrowing as unchecked (which we do for the new `WithFailures` code below). But that could potentially break existing user code if users catch `PipelineExecutionException` and check the cause. That seems unlikely, so perhaps switching to a `Contextful` to avoid rethrowing is an option.

In lieu of that, I've changed this to the more specific `UncheckedIOException`, which inherits from `RuntimeException` so is a safer change.

This all applies to `ParseJsons` too.